src/evaluateXPath.ts

```diff
@@ -184,19 +199,23 @@ const evaluateXPath = <TNode extends Node, TReturnType extends keyof IReturnType
 	}
 
 	try {
-		markXPathStart(selector);
+		if (typeof selector === 'string') {
+			markXPathStart(selector);
+		}
 		const rawResults = expression.evaluateMaybeStatically(dynamicContext, executionParameters);
 		const toReturn = convertXDMReturnValue<TNode, TReturnType>(
-			selector,
+			evaluableExpressionToString(selector),
```

Comment:
It is used in ConvertXDMReturnValue.ts to print errors instead of throwing them. Therefore I've left the function in a separate file. I did fixed the cases where it was just leading to printAndRethrowError.